src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h

```diff
@@ -0,0 +1,416 @@
+     *
+     *   @return Success if the stream transport modification is successful; otherwise, the command SHALL be rejected with an
+     * appropriate error.
+     */
+    virtual Protocols::InteractionModel::Status ModifyPushTransport(const uint16_t connectionID,
+                                                                    const TransportOptionsStruct & transportOptions) = 0;
+
+    /**
+     *   @brief Handle Command Delegate for Stream transport modification.
+     *
+     *   @param connectionIDList [in]       represent the list of connectionIDs for which new transport status to apply.
+     *   @param transportStatus  [in]       represents the updated status of the connection(s).
+     *
+     *   @return Success if the stream transport status is successfully set; otherwise, the command SHALL be rejected with an
+     * appropriate error.
+     */
+    virtual Protocols::InteractionModel::Status SetTransportStatus(const std::vector<uint16_t> connectionIDList,
+                                                                   TransportStatusEnum transportStatus) = 0;
+    /**
+     *   @brief Handle Command Delegate to request the Node to manually start the specified push transport.
+     *
+     *   @param connectionID  [in]          Indicates the connectionID of the stream transport to set trigger for.
+     *
+     *   @param activationReason [in]       Provide information as to why the transport was started or stopped.
+     *
+     *   @param timeControl   [in]          Configuration to control the life cycle of a triggered transport.
+     *
+     *   @return Success if the stream transport trigger is successful; otherwise, the command SHALL be rejected with an
+appropriate
+     *   error.
+     */
+    virtual Protocols::InteractionModel::Status
+    ManuallyTriggerTransport(const uint16_t connectionID, TriggerActivationReasonEnum activationReason,
+                             const Optional<Structs::TransportMotionTriggerTimeControlStruct::DecodableType> & timeControl) = 0;
+
+    /**
+     * @brief Validates the url
+     * @param[in] url The url to validate
+     *
+     * @return boolean value true if the url is valid else false.
+     */
+    virtual bool validateUrl(std::string url) = 0;
+
+    /**
+     * @brief Validates the bandwidth requirement against the camera's resource management
+     *
+     * The implementation SHALL ensure:
+     *  - The requested stream usage (streamUsage) is allowed given the current allocation of
+     *    camera resources (e.g. CPU, memory, network bandwidth).
+     *
+     * @param[in] streamUsage    The desired usage type for the stream (e.g. live view, recording, etc.).
+     * @param[in] videoStreamId  Optional identifier for the requested video stream.
+     * @param[in] audioStreamId  Optional identifier for the requested audio stream.
+     *
+     * @return CHIP_ERROR CHIP_NO_ERROR if the stream usage is valid; an appropriate error code otherwise.
```

Comment:
What makes an error code "appropriate" here?  As an implementor of this API, how do I pick?